personalcapital.js

```diff
@@ -5,14 +5,20 @@ function setPrices(holdings, callback) {
         if (this.readyState === 4) {
             var fixed = [];
             var data = JSON.parse(this.responseText);
-            data.forEach(function(c) {
-                holdings.forEach(function(h) {
+            holdings.forEach(function(h) {
+                var found = false;
+                data.forEach(function(c) {
                     //Trim any special characters
                     if (h.ticker && h.ticker.toLowerCase().match(/[a-z-]+/g)[0] === c.id) {
                         h.price = +c.price_usd;
                         fixed.push(h);
-                    }
+                        found = true;
+                    }    
                 });
+                // Push original in order to get erc20 balance updates
+                if (!found && h.description && h.description.toLowerCase().slice(0, 8) === "erc20:0x") {
```

Comment:
we could probably skip the loop over the coin list here and put this above the `data.forEach` call. I doubt anyone will put this prefix into their description field. Then you don't need the `found` variable.

Reply:
Good call!

Reply:
The reason that we need found is that we want to get the price if possible from coinmarketcap. Otherwise, we still still want the balance.